### PR TITLE
feat(PUPIL-166): display attribution for quiz images

### PR DESCRIPTION
**Pull request by carlmw**
## Description

Music year: [1951](https://www.youtube.com/watch?v=bjCoKslQOEs)

Displays image attribution when present below the questions of a starter or exit quiz.

## How to test

1. Go to https://deploy-preview-2245--oak-web-application.netlify.thenational.academy/pupils/programmes/english-primary-ks2/units/developing-reading-preferences-in-year-5/lessons/developing-reading-preferences-in-year-5-through-appreciation-of-characters
2. Click on "Starter Quiz"
3. You should see attribution text beneath the first question 

### A few other lessons containing examples of attribution text:

https://deploy-preview-2245--oak-web-application.netlify.thenational.academy/pupils/programmes/english-primary-ks2/units/developing-reading-preferences-in-year-6/lessons/developing-reading-preferences-by-appreciating-characters-in-adaptations

https://deploy-preview-2245--oak-web-application.netlify.thenational.academy/pupils/programmes/english-primary-ks2/units/developing-reading-preferences-in-year-6/lessons/developing-reading-preferences-in-year-6-through-text-recommendations

### Containing attribution on question images

https://deploy-preview-2245--oak-web-application.netlify.thenational.academy/pupils/programmes/science-secondary-ks3/units/moving-by-force/lessons/reading-distance-time-graphs

https://deploy-preview-2245--oak-web-application.netlify.thenational.academy/pupils/programmes/english-primary-ks2/units/weather-cold-calm-and-stormy-words/lessons/rich-vocabulary-associated-with-stormy-words

### Desktop
localhost_3000_pupils_programmes_english-primary-ks2_units_developing-reading-preferences-in-year-5_lessons_developing-reading-preferences-in-year-5-through-appreciation-of-characters

Screenshot 2024-02-14 at 12 30 28

### Tablet
localhost_3000_pupils_programmes_english-primary-ks2_units_developing-reading-preferences-in-year-5_lessons_developing-reading-preferences-in-year-5-through-appreciation-of-characters(iPad Air)

### Mobile
localhost_3000_pupils_programmes_english-primary-ks2_units_developing-reading-preferences-in-year-5_lessons_developing-reading-preferences-in-year-5-through-appreciation-of-characters(iPhone 14 Pro Max)

### CODE REVIEW

File: src/components/PupilComponents/QuizAttribution/QuizAttribution.test.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        <QuizAttribution
+          questionData={{
+            ...quizQuestions[0]!,
+            answers: {
+              "multiple-choice": mcqImageAnswers.map((choice, i) => ({
+                ...choice,
+                answer: choice.answer.filter(isImage).map((answer) => ({
+                  ...answer,
+                  image_object: {
+                    ...answer.image_object,
```
Comment (carlmw) [inline src/components/PupilComponents/QuizAttribution/QuizAttribution.test.tsx:47]: Not a fan of this nesting but it's better than copying the whole fixture across. Come utilities for generating complex fixtures wouldn't go amiss in the future.

**Comment by carlmw** [comment]
> @carlmw all good, I was just wondering if we could update the attributions on question images that have multiple to say '**Q1** XXX, **Q2** xxx' etc with a comma added after each one if possible? That is the format of the attributions for multiple images at the bottom of the starter & exit quiz pdf's, sop would be good to match if that isn't too difficult to update in this pr? E.g: Screenshot 2024-02-14 at 12 57 38 pm
> 
> For question stems with an image at the top, that would be the first, then the '**Q1** XXX, **Q2** xxx' etc

@messmike 
🤔 these aren't the images for the questions — they're the images for the answers in an MCQ with images. Would "**A1** XXX, **A2** XXX" be an appropriate treatment?

**Review by kimon-satan — APPROVED** [review]

**Comment by messmike** [comment]
> > @carlmw all good, I was just wondering if we could update the attributions on question images that have multiple to say '**Q1** XXX, **Q2** xxx' etc with a comma added after each one if possible? That is the format of the attributions for multiple images at the bottom of the starter & exit quiz pdf's, sop would be good to match if that isn't too difficult to update in this pr? E.g: Screenshot 2024-02-14 at 12 57 38 pm
> > For question stems with an image at the top, that would be the first, then the '**Q1** XXX, **Q2** xxx' etc
> 
> @messmike 🤔 these aren't the images for the questions — they're the images for the answers in an MCQ with images. Would "**A1** XXX, **A2** XXX" be an appropriate treatment?

Yep, you are totally correct! In that case, just the numbers are probably better. Cheers @carlmw

**Review by messmike — APPROVED** [review]

**Pull request merged**